app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -372,6 +379,10 @@ protected void showNextView() {
     }
 
     private void showNextView(boolean resuming) {
+        if (shouldIgnoreNavigationAction()) {
```

Comment:
Below we block the navigation on "next" instead of "showNextView", is there a reason it makes sense to handle this processing in two different places in the chain, exactly?